packages/sfui/frameworks/react/components/SfScrollable/SfScrollable.tsx

```diff
@@ -0,0 +1,131 @@
+import { useMemo } from 'react';
+import classNames from 'classnames';
+import {
+  polymorphicForwardRef,
+  SfButton,
+  SfIconChevronLeft,
+  SfIconChevronRight,
+  useScrollable,
+  SfScrollableDirection,
+  SfScrollableButtonsPlacement,
+  type SfScrollableProps,
+} from '@storefront-ui/react';
+
+const defaultScrollableTag = 'div';
+
+const SfScrollable = polymorphicForwardRef<typeof defaultScrollableTag, SfScrollableProps>(
+  (
+    {
+      as,
+      direction = SfScrollableDirection.horizontal,
+      buttonsPlacement = SfScrollableButtonsPlacement.block,
+      activeIndex,
+      reduceMotion,
+      drag,
+      onDragChange,
+      onScroll,
+      onPrev,
+      onNext,
+      className,
+      wrapperClassNames,
+      previousDisabled,
+      nextDisabled,
+      style,
+      children,
+      slotPreviousButton,
+      slotNextButton,
+      ...attributes
+    },
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    ref,
+  ) => {
+    const Tag = as || defaultScrollableTag;
+
+    const sliderOptions = useMemo(
+      () => ({
+        direction,
+        activeIndex,
+        reduceMotion,
+        drag,
+        onDragChange,
+        onScroll,
+        onPrev,
+        onNext,
+      }),
+      [direction, activeIndex, reduceMotion, drag, onDragChange, onScroll, onPrev, onNext],
+    );
+
+    const { getContainerProps, state, getNextButtonProps, getPrevButtonProps } = useScrollable(sliderOptions);
+
+    const changeDisabledClass = (isDisabled: boolean) =>
+      isDisabled ? '!ring-gray-200 !text-disabled-500' : '!ring-neutral-500 !text-neutral-500';
+    const previousButton = (...buttonClassName: Parameters<typeof classNames>) =>
+      (slotPreviousButton && <div {...getPrevButtonProps()}>{slotPreviousButton}</div>) || (
+        <SfButton
+          variant="secondary"
+          size="lg"
+          className={classNames(
+            buttonClassName,
+            changeDisabledClass(
+              typeof previousDisabled === 'boolean' ? previousDisabled : getPrevButtonProps().disabled,
+            ),
+          )}
+          square
+          slotPrefix={<SfIconChevronLeft />}
+          {...getPrevButtonProps()}
+          disabled={previousDisabled}
```

Comment:
1. Why didn't used there just `state.hasPrev`, but dedicated prop? Maybe these dedicated props are redundant there when user can just replace a whole component with `slot` / `render prop`.

2. Maybe `useScrollable` hook would not be used internally, but declared by user in a block component?

Reply:
in `getPrevButtonProps` we have `disabled` and `onClick` together so we can pass whole package.
What we give in component is only our opinionated functionality and style, if user feel they indeed can use slots and build everything from scratch, by adding their own buttons with different states `disabled` and other states, then go ahead. If thats is not enough user can use hook directly. Most users and probably your use case will be good component itself, but sure there will be cases when its not enough. 

> Maybe useScrollable hook would not be used internally, but declared by user in a block component?

We are building every hook and then component on top of it, because its simpler to use component for 80% cases, but then having possibility use hook in 20% cases. We use internally hook so we would not need to maintain 2 code with similar/same functionality